### PR TITLE
Promote some util functions to commonMain

### PR DESCRIPTION
**Pull request by swankjesse**
Also migrate some tests that use these Util functions from Java to Kotlin.
If we're going to make OkHttp multiplatform, our tests need to be Kotlin
so we can run them on all supported platforms.

### CODE REVIEW

File: okhttp/src/commonMain/kotlin/okhttp3/internal/-HostnamesCommon.kt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012 The Android Open Source Project
+ * Copyright (C) 2021 Square, Inc.
```
Comment (swankjesse) [inline okhttp/src/commonMain/kotlin/okhttp3/internal/-HostnamesCommon.kt:2]: The GitHub diff suggests a copyright owner change. All of the code in this file is authored by Square.

File: okhttp/src/jvmTest/java/okhttp3/UrlComponentEncodingTester.kt [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.internal.format
+import okio.Buffer
+import okio.ByteString.Companion.encodeUtf8
+import org.junit.jupiter.api.Assertions.fail
+
+/** Tests how each code point is encoded and decoded in the context of each URL component.  */
+internal class UrlComponentEncodingTester private constructor() {
```
Comment (swankjesse) [inline okhttp/src/jvmTest/java/okhttp3/UrlComponentEncodingTester.kt:26]: Working towards making HttpUrl multiplatform.

File: okhttp/src/jvmTest/java/okhttp3/internal/publicsuffix/PublicSuffixDatabaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+import okhttp3.internal.toCanonicalHost
+import okio.Buffer
+import okio.GzipSource
+import okio.buffer
+import okio.source
+import okio.use
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class PublicSuffixDatabaseTest {
```
Comment (swankjesse) [inline okhttp/src/jvmTest/java/okhttp3/internal/publicsuffix/PublicSuffixDatabaseTest.kt:27]: Migrating the database itself is annoying due to its use of Java resource loading! Finally time to address that in Okio??

**Comment by yschimke** [comment]
This is a tough review, I'm assuming most of it is mechanical and our strong test suite is the real reviewer here.

**Review by yschimke — APPROVED** [review]

**Comment by swankjesse** [comment]
Yeah, this is mechanical. I did a few long slow passed fixing up machine-translated code, but nothing interesting!

**Review by maldito-dot — COMMENTED** [review]
@maldito-dot

**Pull request merged**